nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,344 @@
+
+## Retrieving Custom Field Data
+
+Custom fields augment an existing model so retrieving custom field values is different from native fields. All custom field data is stored as a dictionary in the model field named `_custom_field_data` but there is a property named `cf` to make accessing this field easier. Example:
+
+### Retrieve Custom Field Data in Nautobot Shell
+
+```py
+# retrieve all custom field data
+>>> device.cf
+{'date_custom_field': '1970-01-01', 'text_custom_field': 'This is the custom field value', 'boolean_custom_field': True, 'integer_custom_field': 12345, 'grouped_custom_field_1': '', 'grouped_custom_field_2': '', 'grouped_custom_field_3': ''}
+
+# retrieve a single field
+>>> device.cf.get("date_custom_field")
+'1970-01-01'
+```
+
+### Retrieve Custom Field Data in the Rest API
+
+Custom fields are returned in the API for all supported models in the `custom_fields` key:
+
+`GET http://localhost:8080/api/dcim/devices/ffd8df99-6d1a-41c3-b19f-b8357eefc481/`
+
+<!-- markdownlint-disable MD033 -->
+<details>
+<summary>View API Results</summary>
+
+```json
+{
+  "id": "ffd8df99-6d1a-41c3-b19f-b8357eefc481",
+  ...
+  "custom_fields": {
+    "boolean_custom_field": true,
+    "date_custom_field": "1970-01-01",
+    "grouped_custom_field_1": "",
+    "grouped_custom_field_2": "",
+    "grouped_custom_field_3": "",
+    "integer_custom_field": 12345,
+    "text_custom_field": "This is the custom field value"
+  }
+}
+```
+
+</details>
+
+### GraphQL
+
+#### Retrieve Data for a Custom Field in GraphQL
+
+Individual custom fields can be retrieved in GraphQL queries by using the `cf_<fieldname>` field name format:
+
+```graphql
+{
+  devices {
+    cf_text_custom_field
+    name
+    id
+  }
+}
+```
+
+<details>
+<summary>View GraphQL Results</summary>
+
+```json
+{
+  "data": {
+    "devices": [
+      {
+        "name": "phx-leaf1-1-1",
+        "cf_text_custom_field": "This is the custom field value",
+        "id": "8bd9ed2b-3774-4806-9d17-c9f21f2c73e4"        
+      },
+      {
+        "name": "stl-leaf1-2-1",
+        "cf_text_custom_field": "New value",
+        "id": "b22bb7f4-6a6d-4426-9d27-5dcb0471ed2a"        
+      },
+      {
+        "name": "Test Device",
+        "cf_text_custom_field": "Some other value",
+        "id": "ffd8df99-6d1a-41c3-b19f-b8357eefc481"
+      }
+    ]
+  }
+}
+```
+
+</details>
+
+#### Retrieve Data For All Custom Fields in GraphQL
+
+All custom field data can be retrieved in GraphQL queries by using the `_custom_field_data` field:
+
+```graphql
+{
+  devices(id:"8bd9ed2b-3774-4806-9d17-c9f21f2c73e4") {
+    name
+    id
+    _custom_field_data
+  }
+}
+```
+
+<details>
+<summary>View GraphQL Results</summary>
+
+```json
+{
+  "data": {
+    "devices": [
+      {
+        "name": "phx-leaf1-1-1",
+        "id": "8bd9ed2b-3774-4806-9d17-c9f21f2c73e4",        
+        "_custom_field_data": {
+          "date_custom_field": null,
```

Comment:
Just checking - is `null` the expected value here?

Reply:
It's valid but I'll change it to avoid confusion